extensions/AtlasMongoDb/AtlasMongoDb/MongoDbVectorMemory.cs

```diff
@@ -0,0 +1,472 @@
+            await collection.DeleteManyAsync(x => x["index"] == normalizedIndexName, cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+        else
+        {
+            var collectionName = this.GetCollectionName(index);
+            await this._utils.DeleteIndicesAsync(collectionName).ConfigureAwait(false);
+            await this.Database.DropCollectionAsync(collectionName, cancellationToken).ConfigureAwait(false);
+        }
+
+        var collectionIndexList = this.Database.GetCollection<BsonDocument>(GetIndexListCollectionName());
+        await collectionIndexList.DeleteOneAsync(x => x["_id"] == index, cancellationToken: cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<IEnumerable<string>> GetIndexesAsync(CancellationToken cancellationToken = default)
+    {
+        //we load index from the index list collection
+        var collection = this.Database.GetCollection<BsonDocument>(GetIndexListCollectionName());
+        var cursor = await collection.FindAsync(Builders<BsonDocument>.Filter.Empty, cancellationToken: cancellationToken).ConfigureAwait(false);
+        return cursor.ToEnumerable(cancellationToken: cancellationToken).Select(x => x["_id"].AsString).ToImmutableArray();
+    }
+
+    public Task DeleteAsync(string index, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        var collection = this.GetCollectionFromIndexName(index);
+        return collection.DeleteOneAsync(x => x["_id"] == record.Id, cancellationToken: cancellationToken);
+    }
+
+    public async IAsyncEnumerable<MemoryRecord> GetListAsync(
+        string index,
+        ICollection<MemoryFilter>? filters = null,
+        int limit = 1,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (limit <= 0)
+        {
+            limit = 10;
+        }
+
+        //need to create a search query and execute it
+        var normalizedIndexName = NormalizeIndexName(index);
+        var conditions = new BsonArray();
+        this.ConvertsFilterToConditions(normalizedIndexName, filters, conditions);
+        BsonDocument[] pipeline = await this.CreatePipelineAsync(normalizedIndexName, conditions, limit, null).ConfigureAwait(false);
+
+        var collection = this.GetCollectionFromIndexName(index);
+
+        using var cursor = await collection.AggregateAsync<BsonDocument>(pipeline, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+        IEnumerable<BsonDocument> documents = cursor.ToEnumerable(cancellationToken: cancellationToken).Take(limit);
+        foreach (var document in documents)
+        {
+            yield return FromBsonDocument(document, withEmbeddings);
+        }
+    }
+
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetSimilarListAsync(
+        string index,
+        string text,
+        ICollection<MemoryFilter>? filters = null,
+        double minRelevance = 0,
```

Comment:
looks like this is never used, does the method filter out records by relevance if a user passes a value greater than 0?

Reply:
We have a problem here, I can try to implement this but atlas does not seems to filter for cosine similarity value, because it internally normalize the score ...  The only quick way that I can do is filtering client side, stopping the enumeration when the value falls below the relevance.

Also many vectror db like pinecone and also atlas mongodb can compute similiaryt with standard euclidean distance or cosine similarity or dot product (in case of unary vectors) so I'm not sure how to interpret that minRelevance score. If you ensure me that it is always pure cosine similiarity I can easily filter out during enumeration. Thanks

Reply:
Thanks, I will take a look soon. About similarity, currently the expectation is to always use Cosine Similarity and to have the same value across all the engines. There was some thought to include other metrics in future, via configuration, but the default should be the same, so users can switch storage without having having to think about the math :-)

Reply:
I completely agree, sadly mongodb does some normalization, but in the end you can still filter in memory. 😃

Reply:
Also I recalculate pure cosine similarity in memory after query, from this point of view the client will have consistent numbers.